core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCommits.java

```diff
@@ -293,4 +309,59 @@ public void testMergeAppend() throws Exception {
     Assert.assertEquals("Current snapshot should contain 1 merged manifest",
         1, metadata.currentSnapshot().manifests().size());
   }
+
+  @Test
+  public void testRenameFailures() throws Exception {
+    Assert.assertTrue("Should create v1 metadata",
+        version(1).exists() && version(1).isFile());
+    Assert.assertFalse("Should not create v2 or newer versions",
+        version(2).exists());
+    assertTrue(table instanceof BaseTable);
+    final BaseTable bt = (BaseTable) table;
+    // use v1 metafile as the test commit destination
+    final TableMetadata meta1 = bt.operations().current();
+
+    // create v2 metafile as base
```

Comment:
Why does this test depend on v2 metadata?

Reply:
It doesn't depend on it. I just found this to be the easiest way to create another valid metadata file to call `commit` with (that will reach the rename part). Added some comments into the test for this too.
LMK if there's a better / cleaner way though.